### ISSUE TITLE
ERROR: /apollo/modules/dreamview/backend/point_cloud/BUILD:5:1: C++ compilation of rule '//modules/dreamview/backend/point_cloud:point_cloud_updater' failed (Exit 4)

### ISSUE THREAD

**Issue by xqitong** [ApolloAuto/apollo#4162]
~/myapp/apollo-master(master*) » bash docker/scripts/dev_into.sh                                                                                                        xqit@ubuntu
xqit@in_dev_docker:/apollo$ bash apollo.sh build
System check passed. Build continue ...
[WARNING] ESD CAN library supplied by ESD Electronics does not exist. If you need ESD CAN, please refer to third_party/can_card_library/esd_can/README.md.
[INFO] Start building, please wait ...
INFO: Reading 'startup' options from /apollo/tools/bazel.rc: --batch_cpu_scheduling
...........................
[INFO] Building on x86_64...
INFO: Reading 'startup' options from /apollo/tools/bazel.rc: --batch_cpu_scheduling
INFO: Analysed 1472 targets (12 packages loaded).
INFO: Found 1472 targets...
ERROR: /apollo/modules/dreamview/backend/point_cloud/BUILD:5:1: C++ compilation of rule '//modules/dreamview/backend/point_cloud:point_cloud_updater' failed (Exit 4)
gcc: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.8/README.Bugs> for instructions.
INFO: Elapsed time: 16919.951s, Critical Path: 16260.37s
FAILED: Build did NOT complete successfully
============================
[ERROR] Build failed!
[INFO] Took 16934 seconds
============================

I built it twice, but failed. Is that my system memory is too small or any other problems?

**Comment by siyangy** [comment]
Yeah, this is due to insufficient system memory. Please make sure your system has at least 2G free memory to compile.
